### PR TITLE
Dev/streams

### PR DESCRIPTION
**Pull request by duplexsystem**

### CODE REVIEW

**Comment by Titaniumtown** [comment]
What does this do?

File: patches/server/0034-IntelliJ-Remove-Streams.patch [hunk 1]
```diff
@@ -0,0 +1,603 @@
...
+
+diff --git a/src/main/java/net/minecraft/server/BehaviorGate.java b/src/main/java/net/minecraft/server/BehaviorGate.java
+index 46e910581210421c8699637431804dc2f43eb4a6..945c7d0957e5fcdd4dc3b3ac95e274bdee23bf3f 100644
+--- a/src/main/java/net/minecraft/server/BehaviorGate.java
++++ b/src/main/java/net/minecraft/server/BehaviorGate.java
+@@ -19,9 +19,9 @@ public class BehaviorGate<E extends EntityLiving> extends Behavior<E> {
+         this.b = set;
+         this.c = behaviorgate_order;
+         this.d = behaviorgate_execution;
+-        list.forEach((pair) -> {
```
Comment (Proximyst) [inline patches/server/0034-IntelliJ-Remove-Streams.patch:16]: This isn't even a stream.

https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/tip/src/share/classes/java/util/ArrayList.java#l1226

File: patches/server/0034-IntelliJ-Remove-Streams.patch [hunk 2]
```diff
@@ -0,0 +1,603 @@
...
++            this.e.a(pair.getFirst(), pair.getSecond());
++        }
+     }
+ 
+     @Override
+@@ -62,7 +62,10 @@ public class BehaviorGate<E extends EntityLiving> extends Behavior<E> {
+         });
+         BehaviorController behaviorcontroller = e0.getBehaviorController();
+ 
+-        this.b.forEach(behaviorcontroller::removeMemory); // Paper - decomp fix
```
Comment (Proximyst) [inline patches/server/0034-IntelliJ-Remove-Streams.patch:29]: Not a stream.

https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/tip/src/share/classes/java/lang/Iterable.java#l72

File: patches/server/0034-IntelliJ-Remove-Streams.patch [hunk 3]
```diff
@@ -0,0 +1,603 @@
...
+     @Override
+diff --git a/src/main/java/net/minecraft/server/ChunkConverter.java b/src/main/java/net/minecraft/server/ChunkConverter.java
+index 5366314e5f889b5b8d7740bbd0f024d9b7b9d643..a238b269eec03e2d81688907329a04bbb07943cf 100644
+--- a/src/main/java/net/minecraft/server/ChunkConverter.java
++++ b/src/main/java/net/minecraft/server/ChunkConverter.java
+@@ -70,9 +70,9 @@ public class ChunkConverter {
+ 
+         World world = chunk.getWorld();
+ 
+-        ChunkConverter.g.forEach((chunkconverter_a) -> {
```
Comment (Proximyst) [inline patches/server/0034-IntelliJ-Remove-Streams.patch:45]: Not a stream.

https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/tip/src/share/classes/java/lang/Iterable.java#l72

File: patches/server/0034-IntelliJ-Remove-Streams.patch [hunk 4]
```diff
@@ -0,0 +1,603 @@
...
++
++import java.util.*;
+ import java.util.function.Predicate;
+ import java.util.stream.Collectors;
+ import java.util.stream.Stream;
+@@ -27,28 +23,57 @@ public enum EnumDirection implements INamable {
+     private final EnumDirection.EnumAxisDirection l;
+     private final BaseBlockPosition m;
+     private static final EnumDirection[] n = values(); private static final EnumDirection[] ALL = n;// Yatopia OBF HELPER
+-    private static final Map<String, EnumDirection> o = (Map) Arrays.stream(EnumDirection.n).collect(Collectors.toMap(EnumDirection::m, (enumdirection) -> {
```
Comment (Proximyst) [inline patches/server/0034-IntelliJ-Remove-Streams.patch:109]: Streams for what is ran once for rather small sets of data is fine. This isn't a performance issue *at all*.

File: patches/server/0034-IntelliJ-Remove-Streams.patch [hunk 5]
```diff
@@ -0,0 +1,603 @@
...
+         private EnumAxis(String s) {
+diff --git a/src/main/java/net/minecraft/server/EnumProtocol.java b/src/main/java/net/minecraft/server/EnumProtocol.java
+index ec3c8ec6a7c0a835d0bab9872d0ead68b2ff54df..b3cee9bc82aacba99f91ffb4c955ed5321e8250f 100644
+--- a/src/main/java/net/minecraft/server/EnumProtocol.java
++++ b/src/main/java/net/minecraft/server/EnumProtocol.java
+@@ -65,15 +65,17 @@ public enum EnumProtocol {
+             }
+ 
+             EnumProtocol.e[k - -1] = enumprotocol;
+-            enumprotocol.h.forEach((enumprotocoldirection, enumprotocol_a) -> {
```
Comment (Proximyst) [inline patches/server/0034-IntelliJ-Remove-Streams.patch:215]: Not a stream.

https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/tip/src/share/classes/java/util/Map.java#l618

File: patches/server/0034-IntelliJ-Remove-Streams.patch [hunk 6]
```diff
@@ -0,0 +1,603 @@
...
+     }
+diff --git a/src/main/java/net/minecraft/server/PlayerChunkMap.java b/src/main/java/net/minecraft/server/PlayerChunkMap.java
+index 74998ea20176c2753a0ebb03833f13f3f58a186a..7702169fcf0c2cf72f19ef9472ac435cea05cd21 100644
+--- a/src/main/java/net/minecraft/server/PlayerChunkMap.java
++++ b/src/main/java/net/minecraft/server/PlayerChunkMap.java
+@@ -546,13 +546,13 @@ public class PlayerChunkMap extends IChunkLoader implements PlayerChunk.d {
+         if (viewDistance > 6) {
+             ChunkCoordIntPair front7 = player.getChunkInFront(7);
+             pos.setValues(front7.x << 4, 0, front7.z << 4);
+-            MCUtil.getSpiralOutChunks(pos, 3).forEach(coord -> {
```
Comment (Proximyst) [inline patches/server/0034-IntelliJ-Remove-Streams.patch:241]: Not a stream.

https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/tip/src/share/classes/java/util/ArrayList.java#l1226

File: patches/server/0034-IntelliJ-Remove-Streams.patch [hunk 7]
```diff
@@ -0,0 +1,603 @@
...
+         if (priorities.isEmpty()) return;
+diff --git a/src/main/java/net/minecraft/server/Reputation.java b/src/main/java/net/minecraft/server/Reputation.java
+index dffac0dc85fe7d093c58663f91ad687a6be50ad5..5f6a4281204c60f059437e9f04725afa9ff0578d 100644
+--- a/src/main/java/net/minecraft/server/Reputation.java
++++ b/src/main/java/net/minecraft/server/Reputation.java
+@@ -100,14 +100,14 @@ public class Reputation {
+     public void a(Reputation reputation, Random random, int i) {
+         Collection<Reputation.b> collection = reputation.a(random, i);
+ 
+-        collection.forEach((reputation_b) -> {
```
Comment (Proximyst) [inline patches/server/0034-IntelliJ-Remove-Streams.patch:262]: Not a stream.

https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/tip/src/share/classes/java/lang/Iterable.java#l72

File: patches/server/0034-IntelliJ-Remove-Streams.patch [hunk 8]
```diff
@@ -0,0 +1,603 @@
...
+     }
+diff --git a/src/main/java/net/minecraft/server/TickListChunk.java b/src/main/java/net/minecraft/server/TickListChunk.java
+index 8af1229c3da63a838b0bec1cafde1e41cf5cf551..7ee8534bc656e117fa2a21a39693d53ac8a776b8 100644
+--- a/src/main/java/net/minecraft/server/TickListChunk.java
++++ b/src/main/java/net/minecraft/server/TickListChunk.java
+@@ -84,9 +84,9 @@ public class TickListChunk<T> implements TickList<T> {
+     }
+ 
+     public void a(TickList<T> ticklist) {
+-        this.a.forEach((ticklistchunk_a) -> {
```
Comment (Proximyst) [inline patches/server/0034-IntelliJ-Remove-Streams.patch:320]: Not a stream.

https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/tip/src/share/classes/java/util/ArrayList.java#l1226

File: patches/server/0034-IntelliJ-Remove-Streams.patch [hunk 9]
```diff
@@ -0,0 +1,603 @@
...
+     static class a<T> {
+diff --git a/src/main/java/net/minecraft/server/VillagePlaceSection.java b/src/main/java/net/minecraft/server/VillagePlaceSection.java
+index 3aaba7fcaf83a491b9da60d156b7f2379f5a6aeb..5151876951d0089fa2ae229c7fee6e418fea3636 100644
+--- a/src/main/java/net/minecraft/server/VillagePlaceSection.java
++++ b/src/main/java/net/minecraft/server/VillagePlaceSection.java
+@@ -52,7 +52,9 @@ public class VillagePlaceSection {
+         this.c = Maps.newHashMap();
+         this.d = runnable;
+         this.e = flag;
+-        list.forEach(this::a);
```
Comment (Proximyst) [inline patches/server/0034-IntelliJ-Remove-Streams.patch:336]: Not a stream.

https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/tip/src/share/classes/java/util/ArrayList.java#l1226

File: patches/server/0034-IntelliJ-Remove-Streams.patch [hunk 10]
```diff
@@ -0,0 +1,603 @@
...
+     public Stream<VillagePlaceRecord> a(Predicate<VillagePlaceType> predicate, VillagePlace.Occupancy villageplace_occupancy) {
+diff --git a/src/main/java/net/minecraft/server/VillagePlaceType.java b/src/main/java/net/minecraft/server/VillagePlaceType.java
+index 1fb87bf395ced15bc3ef383dae46a42c5e4d2547..5be0e3485bb7a398f4abbaf1a3b6fc57e827aa8e 100644
+--- a/src/main/java/net/minecraft/server/VillagePlaceType.java
++++ b/src/main/java/net/minecraft/server/VillagePlaceType.java
+@@ -115,13 +115,13 @@ public class VillagePlaceType {
+     }
+ 
+     private static VillagePlaceType a(VillagePlaceType villageplacetype) {
+-        villageplacetype.C.forEach((iblockdata) -> {
```
Comment (Proximyst) [inline patches/server/0034-IntelliJ-Remove-Streams.patch:351]: Not a stream.

https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/tip/src/share/classes/java/lang/Iterable.java#l72

File: patches/server/0034-IntelliJ-Remove-Streams.patch [hunk 11]
```diff
@@ -0,0 +1,603 @@
...
+ 
+diff --git a/src/main/java/net/minecraft/server/WeightedList.java b/src/main/java/net/minecraft/server/WeightedList.java
+index 5d9d58411f2fad9d5da703f964d269b4a7c2b205..0879d6f82803e7da963b029eb1ff6c190ab02b8a 100644
+--- a/src/main/java/net/minecraft/server/WeightedList.java
++++ b/src/main/java/net/minecraft/server/WeightedList.java
+@@ -50,7 +50,9 @@ public class WeightedList<U> {
+     public WeightedList<U> a(Random random) {
+         // Paper start - make concurrent safe, work off a clone of the list
+         List<WeightedList.a<U>> list = isUnsafe ? new java.util.ArrayList<WeightedList.a<U>>(this.list) : this.list;
+-        list.forEach((weightedlist_a) -> weightedlist_a.a(random.nextFloat()));
```
Comment (Proximyst) [inline patches/server/0034-IntelliJ-Remove-Streams.patch:373]: Not a stream.

https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/tip/src/share/classes/java/util/ArrayList.java#l1226

File: patches/server/0034-IntelliJ-Remove-Streams.patch [hunk 12]
```diff
@@ -0,0 +1,603 @@
...
+     }
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+index adf918fd757fe3147f897de3ade64a9adf1d3203..3e175ba71086ee587e86a1b732ea3297772e6b05 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+@@ -925,7 +925,9 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
+     @Override
+     public Set<NamespacedKey> getDiscoveredRecipes() {
+         ImmutableSet.Builder<NamespacedKey> bukkitRecipeKeys = ImmutableSet.builder();
+-        getHandle().B().a.forEach(key -> bukkitRecipeKeys.add(CraftNamespacedKey.fromMinecraft(key)));
```
Comment (Proximyst) [inline patches/server/0034-IntelliJ-Remove-Streams.patch:418]: Not a stream.

https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/tip/src/share/classes/java/lang/Iterable.java#l72

File: patches/server/0034-IntelliJ-Remove-Streams.patch [hunk 13]
```diff
@@ -0,0 +1,603 @@
...
+     @Override
+diff --git a/src/test/java/com/destroystokyo/paper/entity/ai/VanillaMobGoalTest.java b/src/test/java/com/destroystokyo/paper/entity/ai/VanillaMobGoalTest.java
+index 232c382b586b0812c9f7161565c0d382177adf7d..ac1c7160d45f88008bba68184045e79b5e647fa9 100644
+--- a/src/test/java/com/destroystokyo/paper/entity/ai/VanillaMobGoalTest.java
++++ b/src/test/java/com/destroystokyo/paper/entity/ai/VanillaMobGoalTest.java
+@@ -43,11 +43,15 @@ public class VanillaMobGoalTest {
+             classes = scanResult.getSubclasses("net.minecraft.server.PathfinderGoal").loadClasses();
+         }
+ 
+-        List<GoalKey<?>> vanillaNames = classes.stream()
```
Comment (Proximyst) [inline patches/server/0034-IntelliJ-Remove-Streams.patch:486]: This is literally a test. Minuscule performance gains do not matter.

File: patches/server/0034-IntelliJ-Remove-Streams.patch [hunk 14]
```diff
@@ -0,0 +1,603 @@
...
++            }
++        }
+ 
+         List<GoalKey<?>> missingFromAPI = new ArrayList<>(vanillaNames);
+         missingFromAPI.removeAll(keys);
+@@ -66,13 +70,17 @@ public class VanillaMobGoalTest {
+         }
+         if (missingFromVanilla.size() != 0) {
+             System.out.println("Missing from vanilla: ");
+-            missingFromVanilla.forEach(System.out::println);
```
Comment (Proximyst) [inline patches/server/0034-IntelliJ-Remove-Streams.patch:507]: This is literally a test. Minuscule performance gains do not matter. Also not a stream.

File: patches/server/0034-IntelliJ-Remove-Streams.patch [hunk 15]
```diff
@@ -0,0 +1,603 @@
...
++            for (GoalKey<?> goalKey : missingFromVanilla) {
++                System.out.println(goalKey);
++            }
+             shouldFail = true;
+         }
+ 
+         if (deprecated.size() != 0) {
+             System.out.println("Deprecated (might want to remove them at some point): ");
+-            deprecated.forEach(System.out::println);
```
Comment (Proximyst) [inline patches/server/0034-IntelliJ-Remove-Streams.patch:516]: This is literally a test. Minuscule performance gains do not matter. Also not a stream.

File: patches/server/0034-IntelliJ-Remove-Streams.patch [hunk 16]
```diff
@@ -0,0 +1,603 @@
...
+         if (shouldFail) Assert.fail("See above");
+diff --git a/src/test/java/org/bukkit/MaterialTest.java b/src/test/java/org/bukkit/MaterialTest.java
+index efac35cb7b5931c21d42306009ff6e92c2260677..caa977d75f2ad909d5b335ee69dd24a4bae4a9b9 100644
+--- a/src/test/java/org/bukkit/MaterialTest.java
++++ b/src/test/java/org/bukkit/MaterialTest.java
+@@ -65,10 +65,16 @@ public class MaterialTest extends AbstractTestingBase {
+                 .forEach(expectedOrder::add);
+ 
+         // Then legacy materials in order of ID
+-        Arrays.stream(Material.values())
```
Comment (Proximyst) [inline patches/server/0034-IntelliJ-Remove-Streams.patch:531]: This is literally a test. Minuscule performance gains do not matter.

File: patches/server/0034-IntelliJ-Remove-Streams.patch [hunk 17]
```diff
@@ -0,0 +1,603 @@
...
+ import java.util.Arrays;
++import java.util.HashSet;
+ import java.util.Set;
+ import java.util.stream.Collectors;
+ import net.minecraft.server.EntityTypes;
+@@ -14,7 +15,12 @@ public class EntityTypesTest extends AbstractTestingBase {
+ 
+     @Test
+     public void testMaps() {
+-        Set<EntityType> allBukkit = Arrays.stream(EntityType.values()).filter((b) -> b.getName() != null).collect(Collectors.toSet());
```
Comment (Proximyst) [inline patches/server/0034-IntelliJ-Remove-Streams.patch:564]: This is literally a test. Minuscule performance gains do not matter.

**Review by Proximyst — CHANGES_REQUESTED** [review]
Extremely little which can actually gain any form for performance.

**Comment by duplexsystem** [comment]
@Proximyst This is done automatically I'm not going to go through and filter out which ones provide "Minuscule performance gains" a gain is still a gain.

**Pull request closed without merging**